src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeGeneratorService.kt

```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.barcode
+
+import org.springframework.stereotype.Service
+import kotlin.random.Random
+
+@Service
+class BarcodeGeneratorService {
+  private val maxBarcode = 999_999_999_999
+
+  fun generateBarcode(): String =
```

Comment:
We handle this by making barcode the PK on the `barcodes` table and catering for duplicates when we try to write to that table. I decided not to go for a UUID because there is a user story where the user types in the barcode (scanner not working) so numbers seemed more user friendly.

Reply:
Ah yes, a story where the user manually enters the barcode - yes, a number sequence is much better in that case 👍